engine/src/main/java/io/camunda/zeebe/engine/processing/signal/SignalBroadcastProcessor.java

```diff
@@ -53,17 +57,30 @@ public SignalBroadcastProcessor(
   }
 
   @Override
-  public void processRecord(final TypedRecord<SignalRecord> command) {
+  public void processNewCommand(final TypedRecord<SignalRecord> command) {
+    final long eventKey = keyGenerator.nextKey();
     final var signalRecord = command.getValue();
-    final var key = command.getKey();
-    final var signalName = signalRecord.getSignalNameBuffer();
-    final var eventKey = key > -1 ? key : keyGenerator.nextKey();
 
     stateWriter.appendFollowUpEvent(eventKey, SignalIntent.BROADCASTED, signalRecord);
     responseWriter.writeEventOnCommand(eventKey, SignalIntent.BROADCASTED, signalRecord, command);
+    activateInstance(command);
+
+    commandDistributionBehavior.distributeCommand(eventKey, command);
+  }
+
+  @Override
+  public void processDistributedCommand(final TypedRecord<SignalRecord> command) {
+    stateWriter.appendFollowUpEvent(command.getKey(), SignalIntent.BROADCASTED, command.getValue());
+    activateInstance(command);
```

Comment:
❌ There is no need to activate an instance in a `distributedCommand`. You can see the discussion in the signal logic overview [here](https://github.com/camunda/zeebe/issues/10777)